aspnetcore/Program.cs

```diff
@@ -1,28 +1,56 @@
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
-using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Configuration;
-using Microsoft.Extensions.Hosting;
-using Microsoft.Extensions.Logging;
-
-namespace aspnetcore
+// Create the web application builder.
+var builder = WebApplication.CreateBuilder(args);
+
+// Configure API controllers, and JSON serialization options.
+builder.Services.AddControllers()
+    .AddJsonOptions(options =>
+    {
+        // Prevent cycles due to EF Core navigation properties.
+        options.JsonSerializerOptions.ReferenceHandler = ReferenceHandler.IgnoreCycles;
+        
+        // Don't serialize properties that are null.
+        options.JsonSerializerOptions.DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull;
+    });
+
+// Setup CORS to allow anything.
+builder.Services.AddCors(options => options.AddDefaultPolicy(policy => policy
+    .AllowAnyOrigin()
+    .AllowAnyHeader()
+    .AllowAnyMethod()));
+
+// Add the database context.
+builder.Services.AddDbContext<HardwareStoreContext>(options =>
+{
+    var connectionString = AppUtils.GetConnectionString(builder.Configuration);
+    options.UseNpgsql(connectionString);
+});
+
+// Initialize Sentry.
+builder.WebHost.UseSentry(options =>
 {
-    public class Program
+    // Use the environment variables set by the startup scripts.
+    options.Dsn = Environment.GetEnvironmentVariable("ASPNETCORE_APP_DSN");
+    options.Release = Environment.GetEnvironmentVariable("RELEASE");
+    
+    // Enable some features.
+    options.TracesSampleRate = 1.0;
+    options.AutoSessionTracking = true;

```

Comment:
This will only capture a session when the process starts and end it when it shuts down (client mode). Or was [server mode](https://develop.sentry.dev/sdk/sessions/#server-model) implemented?

I'm aware of some attempts but they never landed:
* https://github.com/getsentry/sentry-dotnet/pull/1144
* https://github.com/getsentry/sentry-dotnet/pull/1162

Should we go without it?

```suggestion
```

Reply:
@bruno-garcia - I brought up server-side session tracking at Client Infra TSC awhile back, and the consensus was to not worry it as a per-request feature, as that's something that Transactions are already covering.  Still, what you are calling "client mode" still actually has benefit for server-based apps, as it will show a release unhealthy if the server process crashes.   I wish we had clearer terminology than "sessions" here, because you're right - it's not the user's session, but rather the lifetime of the application.